Integrations/Forescout/Forescout.py

```diff
@@ -0,0 +1,678 @@
+
+        # Handle error responses gracefully
+        if res.status_code not in {200, 201, 304}:
+            err_msg = 'Error in Forescout Integration API call [{}] - {}'.format(res.status_code, res.reason)
+            try:
+                # Try to parse json error response
+                res_json = res.json()
+                err_msg += '\n{}'.format(res_json.get('message'))
+                return_error(err_msg)
+            except json.decoder.JSONDecodeError:
+                if res.status_code in {400, 401, 501}:
+                    # Try to parse xml error response
+                    resp_xml = ET_PHONE_HOME.fromstring(res.content)
+                    demisto.info(str(res.content))
+                    codes = [child.text for child in resp_xml.iter() if child.tag == 'CODE']
+                    messages = [child.text for child in resp_xml.iter() if child.tag == 'MESSAGE']
+                    err_msg += ''.join([f'\n{code}: {msg}' for code, msg in zip(codes, messages)])
+                return_error(err_msg)
+
+        resp_type = resp_type.casefold()
+        if resp_type == 'json':
+            return res.json()
+        elif resp_type == 'text':
+            return res.text
+        elif resp_type == 'content':
+            return res.content
+        else:
+            return res
+
+    except requests.exceptions.ConnectionError:
+        err_msg = 'Connection Error - Check that the Server URL parameter is correct.'
+        return_error(err_msg)
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Performs basic get request that requires proper authentication
+    """
+    login()
+    demisto.results('ok')
+
+
+def get_host(args):
+    identifier = args.get('identifier', '')
+    fields = args.get('fields', '')
+    url_suffix = '/api/hosts/'
+    id_type, *ident = identifier.split('=')
+    id_type = id_type.casefold()
+    if len(ident) != 1 or id_type not in {'id', 'ip', 'mac'}:
+        err_msg = 'The entered endpoint identifier should be prefaced by the identifier type,' \
+            ' (\'ip\', \'mac\', or \'id\') followed by \'=\' and the actual ' \
+            'identifier, e.g. \'ip=123.123.123.123\'.'  # disable-secrets-detection
+        raise ValueError(err_msg)
+
+    if id_type == 'ip':
+        # API endpoint format - https://{EM.IP}/api/hosts/ip/{ipv4}?fields={prop},..,{prop_n}
+        url_suffix += 'ip/'
+    elif id_type == 'mac':
+        # API endpoint format - https://{EM.IP}/api/hosts/mac/{mac}?fields={prop},..,{prop_n}
+        url_suffix += 'mac/'
+    # if id_type == 'id' don't change url_suffix -it's already in desired format as shown below
+    # API endpoint format - https://{EM.IP}/api/hosts/{obj_ID}?fields={prop},..,{prop_n}
+
+    url_suffix += '='.join(ident)
+    params = {'fields': fields} if fields != '' else None
+    headers = create_web_api_headers()
+    response = http_request('GET', url_suffix, headers=headers, params=params, resp_type='response')
```

Comment:
why are we not asking for `json` here?

Reply:
my mistake. It was left over when I initially was using a function decorator on all of the functions that made `Web API` calls to work with the entity tags they described in the Forescout Web API documentation before I realized it was unnecessary. I'll change it for all of those calls.